src/servers/src/http.rs

```diff
@@ -698,15 +702,26 @@ impl HttpServer {
     fn route_prom<S>(
         prom_handler: PromStoreProtocolHandlerRef,
         prom_store_with_metric_engine: bool,
+        strict_mode: bool,
     ) -> Router<S> {
         let mut router = Router::new().route("/read", routing::post(prom_store::remote_read));
-        if prom_store_with_metric_engine {
+        if prom_store_with_metric_engine && strict_mode {
             router = router.route("/write", routing::post(prom_store::remote_write));
-        } else {
+        } else if prom_store_with_metric_engine && !strict_mode {
+            router = router.route(
+                "/write",
+                routing::post(prom_store::remote_write_without_strict_mode),
+            );
+        } else if !prom_store_with_metric_engine && strict_mode {
             router = router.route(
                 "/write",
                 routing::post(prom_store::route_write_without_metric_engine),
             );
+        } else {
+            router = router.route(
+                "/write",
+                routing::post(prom_store::route_write_without_metric_engine_and_strict_mode),
+            );
```

Comment:
I was swinging by (actually drafted my version but @irenjj was faster 😅).
I'm curious if we can pass `strict_mode` flag to existing handlers using `Extension`? It would help us to reduce duplication.
@v0y4g3r How does that sound to you?

Reply:
Adding redundant functions does seem somewhat unnecessary.

Reply:
> I was swinging by (actually drafted my version but @irenjj was faster 😅). I'm curious if we can pass `strict_mode` flag to existing handlers using `Extension`? It would help us to reduce duplication. @v0y4g3r How does that sound to you?

It is worth noting that `strict_mode` should be a switch determined while database instance starts rather than a runtime flag passes from HTTP params.

Reply:
ahh.. ok, so the only option is have a separate method for `strict_mode`. Got it! Thanks!